src/test/java/ua/com/juja/microservices/keepers/slackbot/model/SlackParsedCommandTest.java

```diff
@@ -1,10 +1,103 @@
 package ua.com.juja.microservices.keepers.slackbot.model;
 
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import ua.com.juja.microservices.keepers.slackbot.exceptions.WrongCommandFormatException;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.containsString;
 import static org.junit.Assert.*;
 
 /**
- * @author
+ * @author Konstantin Sergey
  */
 public class SlackParsedCommandTest {
-    //todo task #11
+    private Map<String, UserDTO> users;
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Before
+    public void setup() {
+        users = new HashMap<>();
+        users.put("@from", new UserDTO("uuid0", "@from"));
+        users.put("@slack1", new UserDTO("uuid1", "@slack1"));
+        users.put("@slack2", new UserDTO("uuid2", "@slack2"));
+    }
+
+    @Test
+    public void getFirstUserInText() {
+        //given
+        String text = "text text @slack1 text";
+        SlackParsedCommand slackParsedCommand = new SlackParsedCommand("@from", text, users);
+        //when
+        UserDTO result = slackParsedCommand.receiveFirstUser();
+        //then
+        assertEquals("UserDTO(uuid=uuid1, slack=@slack1)", result.toString());
+    }
+
+    @Test
+    public void getFirstUserInTextThrowExceptionIfNotUser() {
+        //given
+        String text = "text text text";
+        SlackParsedCommand slackParsedCommand = new SlackParsedCommand("@from", text, users);
+        //then
+        thrown.expect(WrongCommandFormatException.class);
+        thrown.expectMessage(containsString("The text 'text text text' doesn't contains slackName"));
+        //when
+        slackParsedCommand.receiveFirstUser();
+    }
+
+    @Test
+    public void getAllUsers() {
+        //given
+        String text = "text @slack2 text@slack1 text @slack3";
+        SlackParsedCommand slackParsedCommand = new SlackParsedCommand("@from", text, users);
+        //when
+        List<UserDTO> result = slackParsedCommand.receiveAllUsers();
+        //then
+        assertEquals("[UserDTO(uuid=uuid2, slack=@slack2), UserDTO(uuid=uuid1, slack=@slack1)]",
+                result.toString());
```

Comment:
The text "text @slack2 text@slack1 text @slack3" has three slackNames, but result has only two Users